src/fillTank.test.js

```diff
@@ -1,11 +1,133 @@
 'use strict';
 
 describe('fillTank', () => {
-  // const { fillTank } = require('./fillTank');
+  const { fillTank } = require('./fillTank');
 
-  it('should ', () => {
+  it('should be declared', () => expect(fillTank).toBeDefined());
 
+  it('should not return anything', () => {
+    const customer = {
+      money: 3000,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 8,
+      },
+    };
+
+    expect(fillTank(customer, 50)).toBeUndefined();
+  });
+
+  it('should fill the tank with requested amount', () => {
+    const customer = {
+      money: 3000,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 8,
+      },
+    };
+    const fuelPrice = 50;
+
+    fillTank(customer, fuelPrice, 10);
+
+    expect(customer.vehicle.fuelRemains).toBe(18);
+    expect(customer.money).toBe(2500);
+  });
+
+  it('should fill full tank if the amount is not given', () => {
+    const customer = {
+      money: 3000,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 8,
+      },
+    };
+    const fuelPrice = 50;
+
+    fillTank(customer, fuelPrice);
+
+    expect(customer.vehicle.fuelRemains).toBe(40);
+    expect(customer.money).toBe(1400);
+  });
+
+  it('should fill only freeSpace if amount > freeSpace>', () => {
+    const customer = {
+      money: 3000,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 8,
+      },
+    };
+    const fuelPrice = 50;
+    const amount = 35;
+
+    fillTank(customer, fuelPrice, amount);
+
+    expect(customer.vehicle.fuelRemains).toBe(40);
+    expect(customer.money).toBe(1400);
+  });
+
+  it('should fill what the customer can pay for>', () => {
+    const customer = {
+      money: 500,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 8,
+      },
+    };
+    const fuelPrice = 50;
+    const amount = 45;
+
+    fillTank(customer, fuelPrice, amount);
+
+    expect(customer.vehicle.fuelRemains).toBe(18);
+    expect(customer.money).toBe(0);
+  });
+
+  it('should round the poured amount to the tenth>', () => {
+    const customer = {
+      money: 150,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 8,
+      },
+    };
+    const fuelPrice = 38;
+    const amount = 4;
+
+    fillTank(customer, fuelPrice, amount);
+
+    expect(customer.vehicle.fuelRemains).toBe(11.9);
   });
 
-  // write tests here
+  it('should round the price to the nearest hundredth>', () => {
```

Comment:
The test description 'should round the price to the nearest hundredth>' contains a typo with the '>' character. It should be corrected to 'should round the price to the nearest hundredth'.